HW1/P6/P6.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import findspark
+findspark.init('/home/toby/spark')
+
+import pyspark
+
+sc = pyspark.SparkContext(appName="Spark 2")
+sc.setLogLevel('WARN') 
+
+myfile = "pg100.txt"
+lines = sc.textFile(myfile, 4)
+
+def filter_words(line):
+    words = line.split()
+    def good_word(word):
+        if word.isdigit(): return False
+        if word.upper()==word: return False
+        tmp = word[:-1]
+        if len(word)>0 and word[-1]=="." and tmp.upper()==tmp: return False
+        return True
+
+    return [word for word in words if good_word(word)]
+
+words = lines.flatMap(filter_words).collect()
+print "making trigrams..."
+trigrams = sc.parallelize([[(words[i], words[i+1], words[i+2]), 1.] for i in range(len(words)-2)])
```

Comment:
I used for loop as well. But I saw someone else zip each word with an index, subtract 1 from each index to get 2nd RDD, subtract 1 again from each index to get 3rd RDD. Then join three RDDs to get and RDD of (word1,word2,word3). For loops would work for a small data set like this, but join would probably be better for larger ones.